failback.go

```diff
@@ -158,5 +159,8 @@ func (r *FailBack) startResetTimer() chan struct{} {
 
 // Returns true is the response is considered successful given the options.
 func (r *FailBack) isSuccessResponse(a *dns.Msg) bool {
-	return a == nil || !(r.opt.ServfailError && a.Rcode == dns.RcodeServerFailure)
+	aLen := len(a.Answer)
```

Comment:
If `a` is nil here, this will cause a nil-pointer panic. You can instead just add `len(a.Answer)` where you need it below. Or break out the nil-check first.